source/common/http/conn_manager_impl.cc

```diff
@@ -396,6 +396,25 @@ ConnectionManagerImpl::ActiveStream::~ActiveStream() {
   ASSERT(state_.filter_call_state_ == 0);
 }
 
+void ConnectionManagerImpl::ActiveStream::resetIdleTimer() {
+  if (idle_timer_ != nullptr) {
+    idle_timer_->enableTimer(idle_timeout_ms_);
+  }
+}
+
+void ConnectionManagerImpl::ActiveStream::onIdleTimeout() {
+  connection_manager_.stats_.named_.downstream_rq_idle_timeout_.inc();
+  // If headers have not been sent to the user, send a 408.
+  if (response_headers_ != nullptr) {
+    // TODO(htuch): We could send trailers here with an x-envoy timeout header
+    // or gRPC status code.
+    maybeEndEncode(true);
```

Comment:
Isn't this the clean shutdown route?  I'd expect resetStream().
I think if we do things this way the connection may be reused for a subsequent HTTP/1.1 request.

Reply:
Yeah we need to reset here (optimally w/ a relevant h2 reset code if one exists, not sure). I think this will end up resetting the stream, but I think it will also do some stat/log accounting to call the response complete which we don't want, so I think you will need to go deeper than this function and possibly call doEndStream() directly or something else.